### PR TITLE
BF: github - cycle to the next credential when we fail to obtain organization

### LINKED ISSUE #3946: create-sibling-github fails when organization is specified
```shell
$> datalad create-sibling-github -s gh-neurodebian --github-organization neurodebian afni-data
[ERROR  ] 401 {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'} [Requester.py:__check:286] (BadCredentialsException) 
$> datalad --version
datalad 0.12.0rc6.dev198

```
worked fine for myself although complained about failed auth using token
```
$> datalad create-sibling-github -s gh-neurodebian afni-data            
[WARNING] Authentication failed using a token. 
.: gh-neurodebian(-) [https://github.com/yarikoptic/afni-data.git (git)]
'https://github.com/yarikoptic/afni-data.git' configured as sibling 'gh-neurodebian' for <Dataset path=/home/yoh/deb/gits/pkg-exppsy/afni-data>
```
<details><summary>DataLad 0.12.0rc6.dev198 WTF (configuration, datalad, dataset, dependencies, environment, extensions, git-annex, location, metadata_extractors, python, system)</summary>

# WTF
## datalad 
  - full_version: 0.12.0rc6.dev198-g78c29
  - version: 0.12.0rc6.dev198
## dataset 
  - id: None
  - metadata: <SENSITIVE, report disabled by configuration>
  - path: /home/yoh/deb/gits/pkg-exppsy/afni-data
  - repo: GitRepo
## dependencies 
  - appdirs: 1.4.3
  - boto: 2.49.0
  - cmd:annex: 7.20190819+git2-g908476a9b-1~ndall+1
  - cmd:bundled-git: 2.20.1
  - cmd:git: 2.20.1
  - cmd:system-git: 2.24.0
  - cmd:system-ssh: 8.1p1
  - exifread: 2.1.2
  - git: 3.0.5
  - gitdb: 2.0.6
  - humanize: 0.5.1
  - iso8601: 0.1.11
  - keyring: 18.0.1
  - keyrings.alt: 3.1.1
  - msgpack: 0.5.6
  - mutagen: 1.40.0
  - requests: 2.21.0
  - tqdm: 4.30.0
  - wrapt: 1.11.2
## environment 
  - GIT_PAGER: less --no-init --quit-if-one-screen
  - GIT_PYTHON_GIT_EXECUTABLE: /usr/lib/git-annex.linux/git
  - LANG: en_US.UTF-8
  - PATH: /home/yoh/proj/datalad/datalad-master/venvs/dev3/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin:/usr/sbin:/usr/local/sbin
## git-annex 
  - build flags: 
    - Assistant
    - Webapp
    - Pairing
    - S3
    - WebDAV
    - Inotify
    - DBus
    - DesktopNotify
    - TorrentParser
    - MagicMime
    - Feeds
    - Testsuite
  - dependency versions: 
    - aws-0.20
    - bloomfilter-2.0.1.0
    - cryptonite-0.25
    - DAV-1.3.3
    - feed-1.0.0.0
    - ghc-8.4.4
    - http-client-0.5.13.1
    - persistent-sqlite-2.8.2
    - torrent-10000.1.1
    - uuid-1.3.13
    - yesod-1.6.0
  - key/value backends: 
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
  - operating system: linux x86_64
  - remote types: 
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - hook
    - external
  - supported repository versions: 
    - 5
    - 7
  - upgrade supported from repository versions: 
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
  - version: 7.20190819+git2-g908476a9b-1~ndall+1
## location 
  - path: /home/yoh/deb/gits/pkg-exppsy/afni-data
  - type: dataset
## metadata_extractors 
  - annex: 
    - load_error: None
    - module: datalad.metadata.extractors.annex
    - version: None
  - audio: 
    - load_error: None
    - module: datalad.metadata.extractors.audio
    - version: None
  - datacite: 
    - load_error: None
    - module: datalad.metadata.extractors.datacite
    - version: None
  - datalad_core: 
    - load_error: None
    - module: datalad.metadata.extractors.datalad_core
    - version: None
  - datalad_rfc822: 
    - load_error: None
    - module: datalad.metadata.extractors.datalad_rfc822
    - version: None
  - exif: 
    - load_error: None
    - module: datalad.metadata.extractors.exif
    - version: None
  - frictionless_datapackage: 
    - load_error: None
    - module: datalad.metadata.extractors.frictionless_datapackage
    - version: None
  - image: 
    - load_error: None
    - module: datalad.metadata.extractors.image
    - version: None
  - xmp: 
    - load_error: No module named 'libxmp' [xmp.py:<module>:20]
    - module: datalad.metadata.extractors.xmp
## python 
  - implementation: CPython
  - version: 3.7.5
## system 
  - distribution: debian/bullseye/sid
  - encoding: 
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - max_path_length: 295
  - name: Linux
  - release: 5.4.0-trunk-amd64
  - type: posix
  - version: #1 SMP Debian 5.4-1~exp1 (2019-11-26)
</details>

full traceback
```
$> datalad --dbg create-sibling-github -s gh-neurodebian --github-organization neurodebian afni-data
Traceback (most recent call last):
  File "/home/yoh/proj/datalad/datalad-master/venvs/dev3/bin/datalad", line 8, in <module>
    main()
  File "/home/yoh/proj/datalad/datalad-master/datalad/cmdline/main.py", line 525, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/home/yoh/proj/datalad/datalad-master/datalad/interface/base.py", line 718, in call_from_parser
    ret = cls.__call__(**kwargs)
  File "/home/yoh/proj/datalad/datalad-master/datalad/distribution/create_sibling_github.py", line 200, in __call__
    existing, access_protocol, dryrun)
  File "/home/yoh/proj/datalad/datalad-master/datalad/support/github_.py", line 298, in _make_github_repos
    github_organization):
  File "/home/yoh/proj/datalad/datalad-master/datalad/support/github_.py", line 275, in _gen_github_entity
    yield ses.get_organization(github_organization), cred
  File "/usr/lib/python3/dist-packages/github/MainClass.py", line 263, in get_organization
    "/orgs/" + login
  File "/usr/lib/python3/dist-packages/github/Requester.py", line 275, in requestJsonAndCheck
    return self.__check(*self.requestJson(verb, url, parameters, headers, input, self.__customConnection(url)))
  File "/usr/lib/python3/dist-packages/github/Requester.py", line 286, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.BadCredentialsException: 401 {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'}

> /usr/lib/python3/dist-packages/github/Requester.py(286)__check()
-> raise self.__createException(status, responseHeaders, output)
```

### PR DESCRIPTION
**Pull request by yarikoptic**
Without it we immediately might fail if there is a known token which
does not have necessary permissions.  Typically this would lead to just
a failure like

    github.GithubException.BadCredentialsException: 401 {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'

now we will just go to the next credential while also providing an
informative log message on either we succeeded or not.  So the interaction
would look like

	$> datalad create-sibling-github --github-organization datalad-collection-1 testds
	[WARNING] Having authenticated using token, we failed (401 {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'}) to access information about organization datalad-collection-1. We will try next authentication method (if any left available)
	[INFO   ] Successfully obtained information about organization datalad-collection-1 using UserPassword(name='github', url='https://github.com/login') credential
	.: github(-) [https://github.com/datalad-collection-1/testds.git (git)]
	'https://github.com/datalad-collection-1/testds.git' configured as sibling 'github' for <Dataset path=/tmp/testds>

Closes #3946 

Note: Merge to master would cause conflicts since it RFed many (but not all) relative imports to absolute, and thus merge of test_*py would require resolving them.

### CODE REVIEW

File: datalad/support/github_.py [hunk 1]
```diff
@@ -272,12 +279,25 @@ def _gen_github_entity(
...
                                  github_organization,
                                  exc_str(e)))
+            except gh.BadCredentialsException as e:
+                lgr.warning(
+                    "Having authenticated using %s, we failed (%s) to access "
+                    "information about organization %s. We will try next "
+                    "authentication method (if any left available)",
+                    cred or "token", e, github_organization
+                )
+                continue
```
Comment (yarikoptic) [inline datalad/support/github_.py:300]: this is the "heart" of the fix. The rest is just supportive changes ;)

File: datalad/support/tests/test_github_.py [hunk 2]
```diff
@@ -14,12 +14,31 @@
...
-from ..github_ import get_repo_url
+from ..github_ import (
+    _gen_github_entity,
+    _get_github_cred,
+    get_repo_url,
+)
+
+
+skip_if_no_github_cred = \
+    lambda f: skip_if(cond=not _get_github_cred().is_known)(f)
```
Comment (kyleam) [inline datalad/support/tests/test_github_.py:41]: Is this lambda unnecessary?

Comment (yarikoptic) [reply datalad/support/tests/test_github_.py:41]: ha ha ha -- I was dancing around to make it work and have missed that it should be as simple as you suggested ;)  Thank you for the review -- force pushing!

**Comment by yarikoptic** [comment]
comment was resolved, and the world will become a better place

**Pull request merged**